src/MongoDB.Driver.Core/Core/Compression/Zstd/Zstandard64NativeMethods.cs

```diff
@@ -0,0 +1,241 @@
+        {
+            return __ZSTD_DStreamOutSize.Value();
+        }
+
+        public static IntPtr ZSTD_createDStream()
+        {
+            return __ZSTD_createDStream.Value();
+        }
+
+        public static int ZSTD_maxCLevel()
+        {
+            return __ZSTD_maxCLevel.Value();
+        }
+
+        public static UIntPtr ZSTD_flushStream(IntPtr zcs, [MarshalAs(UnmanagedType.LPStruct)] NativeBufferInfo outputBuffer)
+        {
+            var result = __ZSTD_flushStream.Value(zcs, outputBuffer);
+            ThrowIfError(result);
+            return result;
+        }
+
+        public static UIntPtr ZSTD_endStream(IntPtr zcs, [MarshalAs(UnmanagedType.LPStruct)] NativeBufferInfo outputBuffer)
+        {
+            var result = __ZSTD_endStream.Value(zcs, outputBuffer);
+            ThrowIfError(result);
+            return result;
+        }
+
+        public static UIntPtr ZSTD_freeCStream(IntPtr zcs)
+        {
+            return __ZSTD_freeCStream.Value(zcs);
+        }
+
+        public static UIntPtr ZSTD_freeDStream(IntPtr zds)
+        {
+            return __ZSTD_freeDStream.Value(zds);
+        }
+
+        public static UIntPtr ZSTD_initDStream(IntPtr zds)
+        {
+            return __ZSTD_initDStream.Value(zds);
+        }
+
+        public static UIntPtr ZSTD_decompressStream(IntPtr zds, [MarshalAs(UnmanagedType.LPStruct)] NativeBufferInfo outputBuffer, [MarshalAs(UnmanagedType.LPStruct)] NativeBufferInfo inputBuffer)
+        {
+            var result = __ZSTD_decompressStream.Value(zds, outputBuffer, inputBuffer);
+            ThrowIfError(result);
+            return result;
+        }
+
+        public static UIntPtr ZSTD_initCStream(IntPtr zcs, int compressionLevel)
+        {
+            var result = __ZSTD_initCStream.Value(zcs, compressionLevel);
+            ThrowIfError(result);
+            return result;
+        }
+
+        public static UIntPtr ZSTD_compressStream(IntPtr zcs, [MarshalAs(UnmanagedType.LPStruct)] NativeBufferInfo outputBuffer, [MarshalAs(UnmanagedType.LPStruct)] NativeBufferInfo inputBuffer)
+        {
+            var result = __ZSTD_compressStream.Value(zcs, outputBuffer, inputBuffer);
+            ThrowIfError(result);
+            return result;
+        }
+
+        // private static methods
+        private static Lazy<TDelegate> CreateLazyForDelegate<TDelegate>(string name)
+        {
+            return new Lazy<TDelegate>(() => __libraryLoader.Value.GetDelegate<TDelegate>(name), isThreadSafe: true);
+        }
+
+        private static void ThrowIfError(UIntPtr code)
+        {
+            if (Zstandard64NativeMethods.ZSTD_isError(code))
+            {
+                var errorPtr = Zstandard64NativeMethods.ZSTD_getErrorName(code);
+                var errorMsg = Marshal.PtrToStringAnsi(errorPtr);
+                throw new IOException(errorMsg);
+            }
+        }
+
+        private static bool ZSTD_isError(UIntPtr code)
```

Comment:
this is the only change comparing the current version with the previous. I made these two methods private since they were not used outside of this file